### PR TITLE
WasmPlugin: Fix to update the checksum cache when the module cache hits

### PR DESCRIPTION
**Pull request by ingwonsong**
Currently, when the module cache of WasmPlugin hits, the checksum cache is not updated.
This may cause inefficiency when the same image was referenced by the two different tags as well as is not an intended behavior.

This PR fixes it by adding `touchEntry` to update the checksum cache when the module cache hits.

### CODE REVIEW

File: pkg/wasm/cache_test.go [hunk 1]
```diff
@@ -535,13 +711,28 @@ func TestWasmCache(t *testing.T) {
...
 					t.Errorf("Wasm module cache entry's last access time not updated after get operation, key: %v", *cacheHitKey)
 				}
-				cache.mux.Unlock()
 			}
+
+			if diff := cmp.Diff(cache.modules, c.wantCachedModules,
+				cmpopts.IgnoreFields(cacheEntry{}, "last", "referencingURLs"),
+				cmp.AllowUnexported(cacheEntry{}),
+			); diff != "" {
+				t.Errorf("unexpected module cache is resulted:(+want, -got)\n%v", diff)
```
Comment (douglas-reid) [inline pkg/wasm/cache_test.go:725]:
```suggestion
				t.Errorf("unexpected module cache: (+want, -got)\n%v", diff)
```

File: pkg/wasm/cache_test.go [hunk 2]
```diff
@@ -535,13 +711,28 @@ func TestWasmCache(t *testing.T) {
...
+			}
+
+			if diff := cmp.Diff(cache.checksums, c.wantCachedChecksums,
+				cmp.AllowUnexported(checksumEntry{}),
+			); diff != "" {
+				t.Errorf("unexpected checksums is resulted:(+want, -got)\n%v", diff)
```
Comment (douglas-reid) [inline pkg/wasm/cache_test.go:731]:
```suggestion
				t.Errorf("unexpected checksums: (+want, -got)\n%v", diff)
```

File: pkg/wasm/cache.go [hunk 3]
```diff
@@ -375,17 +396,19 @@ func (c *LocalFileCache) purge() {
...
+				}
+				// The module has not be touched for expiry duration, delete it from the map as well as the local dir.
+				if err := os.Remove(m.modulePath); err != nil {
+					wasmLog.Errorf("failed to purge Wasm module %v: %v", m.modulePath, err)
+				} else {
+					for downloadURL := range m.referencingURLs {
+						delete(c.checksums, downloadURL)
 					}
+					delete(c.modules, k)
+					log.Infof("modules :%v", c.modules)
```
Comment (douglas-reid) [inline pkg/wasm/cache.go:410]: is this extra log line helpful?  should it be `wasmLog.Infof`? or `wasmLog.Debugf`?

**Review by douglas-reid — COMMENTED** [review]
looks reasonable. just one question, really.

File: pkg/wasm/cache.go (hunk 3 above)
Comment (ingwonsong) [reply pkg/wasm/cache.go:410]: Oh.. my bad. will remove it. Thanks!

**Review by douglas-reid — APPROVED** [review]

**Pull request merged**